tools/mason/MasonNew.chpl

```diff
@@ -91,13 +103,112 @@ proc masonNew(args) throws {
   }
 }
 
+/*
+  Starts an interactive session to create a
+  new library project.
+*/
+proc beginInteractiveSession(isInit: bool, defaultPackageName: string) throws {
+  writeln("This is an interactive session to walk you through creating a library ",
+   "project using Mason.\nThe following queries covers the common items required to ",
+   "create the project.\nSuggestions for defaults are also provided which will be ",
+   "considered if no input is given.\n");
+  writeln("Press ^C to quit interactive mode.");
+  var packageName: string;
+  var version: string;
+  var chapelVersion: string;
+  var gotCorrectPackageName = false;
+  var gotCorrectPackageVersion = false;
+  var gotCorrectChapelVersion = false;
+  while(1){
+    try {
+      if gotCorrectPackageName == false {
+        write("Package name ");
+        if isInit == true then write("(" + defaultPackageName + ") ");
+        write(": ");
+        IO.stdout.flush();
+        IO.stdin.readline(packageName);
+        packageName = packageName.strip();
+        if isInit == true && packageName == '' then
+          packageName = defaultPackageName; 
+        if validatePackageName(packageName) == true then
+          gotCorrectPackageName = true;
+      }
+      if gotCorrectPackageVersion == false {
+        write("Package version (0.1.0): ");
+        IO.stdout.flush();
+        IO.stdin.readline(version);
+        version = version.strip();
+        if version == "" then version = "0.1.0";
+        checkVersion(version);
+        gotCorrectPackageVersion = true;
+      }
+      if gotCorrectChapelVersion == false {
+        var currChapelVersion = getChapelVersionStr();
+        write("Chapel version (" + currChapelVersion + "): ");
+        IO.stdout.flush();
+        IO.stdin.readline(chapelVersion);
+        chapelVersion = chapelVersion.strip();
+        if chapelVersion == "" then chapelVersion = currChapelVersion;
+        if chapelVersion == currChapelVersion then gotCorrectChapelVersion = true;
+        else if validateChplVersion(chapelVersion) == true
+        then gotCorrectChapelVersion = true;
+      }
+      if gotCorrectPackageName == true &&
+         gotCorrectPackageVersion == true &&
+         gotCorrectChapelVersion == true  {
+          previewMasonFile(packageName, version, chapelVersion);
+          writeln();
+          write("Is this okay ? (Y/N): ");
+          IO.stdout.flush();
+          var option: string;
+          IO.stdin.readline(option);
+          option = option.strip();
+          if option == "Y" then break;
+          if option == "N" then exit(1);
+      }
+    }
+    catch e: MasonError {
+      writeln(e.message());
+      continue;
+    }
+  }
+  return (packageName, version, chapelVersion);
+}
+
+/* Previews the Mason.toml file that is going to be created */
+proc previewMasonFile(packageName, version, chapelVersion) {
+  const baseToml = '[brick]\n' +
```

Comment:
A multi-line string + `.format()` could be cleaner here , e.g.

```chpl
const baseToml = '''[brick]
name = "%s"
...
[dependencies]
'''.format(packageName, version, chapelVersion);
```

Reply:
If you make this function just return the string, could you re-use it in `makeBasicToml`?